lib/grandpa/grandpa.go

```diff
@@ -258,13 +260,41 @@ func (s *Service) updateAuthorities() error {
 	// setting to 0 before incrementing indicates
 	// the setID has been increased
 	s.state.round = 0
+
+	s.sendTelemetryAuthoritySet()
+
 	return nil
 }
 
 func (s *Service) publicKeyBytes() ed25519.PublicKeyBytes {
 	return s.keypair.Public().(*ed25519.PublicKey).AsBytes()
 }
 
+func (s *Service) sendTelemetryAuthoritySet() {
+	authorityID := s.keypair.Public().Hex()
+	authorities := make([]string, len(s.state.voters))
+	for i, voter := range s.state.voters {
+		authorities[i] = fmt.Sprintf("%d", voter.ID)
+	}
+
+	authoritiesBytes, err := json.Marshal(authorities)
+	if err != nil {
+		logger.Debugf("could not marshal authorities: %s", err)
+		return
+	}
+
+	err = telemetry.GetInstance().SendMessage(
+		telemetry.NewAfgAuthoritySetTM(
+			authorityID,
+			fmt.Sprintf("%d", s.state.setID),
+			string(authoritiesBytes),
+		),
+	)
+	if err != nil {
+		logger.Debugf("problem sending afg.authority_set telemetry message: %s", err)
```

Comment:
Should this be a warning? :thinking: Does this happen often? Same question for [line 646](https://github.com/ChainSafe/gossamer/pull/2043/files#diff-ed0316f87687d07bf600fc53bf4e701766f4f76752c8e3e3e832b1a80e35a08eR646)

Reply:
noot prefers it to be `debug`. This is to not bloat our logs, since telemetry messages get sent often. All such messages are kept as debug.